extensions/amp-ad/0.1/amp-ad-cross-domain-iframe-handler.js

```diff
@@ -107,6 +106,9 @@ export class AmpAdApiHandler {
       // If support render-start, create a race between render-start no-content
       this.adResponsePromise_ = listenForOncePromise(this.iframe_,
         ['render-start', 'no-content'], this.is3p_).then(info => {
+          if (this.timeout_) {
```

Comment:
we should be able to remove this internal state `this.timeout_` by moving the `timeoutPromise` logic here before the `then` block.